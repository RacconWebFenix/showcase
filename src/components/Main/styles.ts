'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #333333;
  color: #fff;
  width: 100%;
  height: 100vh;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  p {
    font-size: 1.6rem;
    margin-bottom: 2rem;
  }
`
