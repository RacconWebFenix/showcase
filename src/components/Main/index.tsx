import * as S from './styles'

const Main = ({ title = 'Showcase Marcos Domingues' }: { title: string }) => (
  <S.Wrapper>
    <h1>{title}</h1>
  </S.Wrapper>
)

export default Main
