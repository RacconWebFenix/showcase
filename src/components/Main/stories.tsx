import { Meta, StoryObj } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen',
    controls: { expanded: true }
  }
} as Meta

export const Template: StoryObj = {}

export const Basic: StoryObj = {
  args: {
    title: 'Showcase Marcos Domingues'
  }
}
