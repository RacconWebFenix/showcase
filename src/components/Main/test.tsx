import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the Heading', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /Showcase Marcos Domingues/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
