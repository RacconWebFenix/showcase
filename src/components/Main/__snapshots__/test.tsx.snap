// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the Heading 1`] = `
.c0 {
  background-color: #333333;
  color: #fff;
  width: 100%;
  height: 100vh;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 p {
  font-size: 1.6rem;
  margin-bottom: 2rem;
}

<main
  class="c0"
>
  <h1>
    Showcase Marcos Domingues
  </h1>
  <p>
    Transformando Códigos em Experiências Visuais.
  </p>
</main>
`;
